File: Cpp/482_License_Key_Formatting.cpp
class Solution {
public:
    string licenseKeyFormatting(string S, int K) {
        string result = "";
        int char_count = 0;
        for(int i=S.size()-1; i>=0; --i){
            if(S[i] == '-') continue;
            if(char_count != 0 && char_count%K == 0) result += '-';
            result += toupper(S[i]);
            ++char_count;
        }
        
        return string(result.rbegin(), result.rend());
    }
};

File: Kotlin/482_License Key_Formatting.kt
class Solution {
    fun licenseKeyFormatting(s: String, k: Int): String {
        val result = StringBuilder()
        
        for (i in s.lastIndex downTo 0) {
            if(s[i] == '-') continue
            if(result.length % (k + 1) == k) {
                result.append('-')
            }
            
            result.append(s[i].toUpperCase())
        }
        
        return result.reversed().toString()
    }
}